build_tools/cmake/iree_setup_toolchain.cmake

```diff
@@ -97,6 +97,14 @@ endif()
 if(IREE_ENABLE_ASAN)
   string(APPEND CMAKE_CXX_FLAGS " -fsanitize=address")
   string(APPEND CMAKE_C_FLAGS " -fsanitize=address")
+
+  # Not technically ASAN, but it guards against similar bugs in accessing
+  # uninitialized memory. See the extensive description in that patch that
+  # originally introduced it:
+  # https://reviews.llvm.org/rG14daa20be1ad89639ec209d969232d19cf698845
+  string(APPEND CMAKE_CXX_FLAGS " -ftrivial-auto-var-init=pattern")
+  string(APPEND CMAKE_C_FLAGS " -ftrivial-auto-var-init=pattern")
```

Comment:
Ah, these should be clang only...

MSVC
```
[build] [1885/3718  48% :: 116.116] Building CXX object llvm-project\lib\Transforms\Scalar\CMakeFiles\LLVMScalarOpts.dir\LoopRotation.cpp.obj
[build] cl : Command line warning D9002 : ignoring unknown option '-ftrivial-auto-var-init=pattern'
```

Reply:
`-fsanitize=address` is fine though?

Reply:
Yeah: https://learn.microsoft.com/en-us/cpp/sanitizers/asan?view=msvc-170 (I think the flags get handled correctly with either `-` or `/` prefixes?)

Reply:
Might need something like this: https://github.com/ScottTodd/iree/commit/db84fed882290cc9503917605b1bdc00c335d080 . I had ASan working on Windows at some point, but it isn't working for me right now (all executables/tests either exit with no output or loop forever).

Reply:
Discussion [on Discord here](https://discord.com/channels/689900678990135345/689957613152239638/934658933333831700) of when we had ASan working on Windows, not sure what changed about my setup.

We could just treat the `IREE_ENABLE_ASAN` option as Linux/macOS-only, at least until we have it working on MSVC/Windows reliably.